app/controllers/auth_controller.rb

```diff
@@ -0,0 +1,21 @@
+class AuthController < ApplicationController
+  def new
+    render :login
+  end
+
+  def login
+    @user = User.find_by(username: params[:user][:username])
+    if @user && @user.password == params[:user][:password]
+      session[:user_id] = @user.id
+      redirect_to ideas_path
+    else
+      @errors = "Username and/or Password invalid, try again"
+      render :'login'
```

Comment:
since you are trying to render the `login` template from the `login` action, you don't need to explicitly state this here. it will automatically do this!